### ISSUE TITLE
✨ Contribution Opportunity: Wallet UX Research

### ISSUE THREAD

**Issue by smartcontracts** [ethereum-optimism/ecosystem-contributions#69]
**What is a Contribution Opportunity?**

A Contribution Opportunity is a concrete task that you can work on to contribute to the Optimism Collective.

---

## Summary

Wallets are the entrypoint into the blockchain ecosystem for most users. It's important to gauge how well these wallets are supporting OP Mainnet and other similar OP Stack based chains. It would be highly valuable to have a detailed report on many different wallets in the ecosystem and how well they are supporting OP Stack based chains.

Examples of feature support we are looking for:
- Proper support for the OP Stack fee mechanism so that the total fee is equal to the L2 execution fee plus the L1 data fee
- Native support for OP Mainnet without needing to add it manually
- Proper ENS resolution using Ethereum mainnet values
- Proper surfacing of NFT and ERC20 assets that a wallet may hold
- Smooth experience when sending transactions and proper handling of any RPC rate limiting issues
- Fast transaction times and UI updates

## Potential Features

- Detailed document describing your experience with the above feature support items for many different popular wallets
- Highlights of key missing features in popular wallets
- Screenshots or screen recordings of issues where appropriate
- Testing on both mobile and desktop platforms

## Impact

- Wallets are how users interact with blockchains, bad wallet experiences significant impact the user experience overall
- Wallet issues will impact all OP Stack chains in the same way, so this has a positive impact for the entire Superchain ecosystem

**Comment by smartcontracts** [comment]
@Sednaoui really good points re: ENS. It seems like ENS already supports resolution of other things via custom records? But really need to introduce a standard around that for it to be viable.

**Comment by Sednaoui** [comment]
That's correct, and the introduction for a standard would clarify how to resolve names based on network/chain id. Then there we would need to coordinate with major libraries used by wallets and dapps like etherjs to support it.

**Comment by Wearelove89j23** [comment]
Okay

**Comment by ngerald51** [comment]
I feel there should be an increase feature on having inbulit wallet security. (i.e Showing the transaction/ payment route before signing a transaction)

**Comment by ngerald51** [comment]
Please find Rabby wallet UX Audit
https://www.notion.so/Rabby-Wallet-074d0cc5e80643dcb3e1c4f8029c5fa1?pvs=4
